File: tests/bridges/scripts/t_declares.py
"""Declares one port of every shape the bridge can describe, and does nothing with them."""

import atp


class Declares(atp.Module):
    name = "py_declares"
    version = (2, 1)

    state_i32 = atp.Input(atp.i32)
    queue_i32 = atp.Input(atp.i32, queue=True, capacity=8, overflow=atp.DROP_INCOMING)
    in_text = atp.Input(atp.text)

    out_i64 = atp.Output(atp.i64)
    out_f64 = atp.Output(atp.f64)
    out_blob = atp.Output(atp.blob)

    gain = atp.Property(atp.f64, 1.5)
    mode = atp.Property(atp.text, "plain", options=("plain", "verbose"))
    transient = atp.Property(atp.bool_, False, persistent=False)

    def iterate(self):
        return atp.IDLE

File: tests/bridges/lua_scripts/t_declares.lua
local atp = require("atp")

local M = atp.module("lua_declares", { 2, 1 })

M.value = atp.input(atp.i32)
M.queued = atp.input(atp.text, { queue = true, capacity = 4, overflow = atp.DROP_INCOMING })
M.flag = atp.input(atp.bool)

M.result = atp.output(atp.i32)
M.label = atp.output(atp.text)

M.factor = atp.property(atp.i32, 2)
M.channels = atp.property(atp.i32, 2, { options = { 1, 2, 6 } })
M.transient = atp.property(atp.text, "x", { persistent = false })

function M:iterate()
    return atp.IDLE
end

return M
